### PR TITLE
Add support for Openstack region

### PR DESCRIPTION
**Pull request by MikaYuoadas**
Hello,

I've added an extra `region` argument to the `__init__()` of `OpenStackLatentWorker`.
We are using this to spawn Worker on different regions.

### CODE REVIEW

**Comment by tardyp** [comment]
Please add some unit tests for your new feature, as well as documentation, and a bullet in the release note

File: master/buildbot/worker/openstack.py [hunk 1]
```diff
@@ -72,7 +73,7 @@ def __init__(self, name, password,
         self.client_version = client_version
         self.novaclient = client.Client(client_version, os_username,
                                         os_password, os_tenant_name,
-                                        os_auth_url)
+                                        os_auth_url, region_name=region)
```
Comment (rayene) [inline master/buildbot/worker/openstack.py]: I would replace `region_name` with a dictionary (e.g. `**client_kwargs`) to cover even more use cases. The `Client` class has many parameters that cannot be used fron the OpenstackLatentWorker.

**Comment by MikaYuoadas** [comment]
Ok, I'll work on adding it.

**Comment by tardyp** [comment]
Oh, and also please submit your pull request against master. not against the stable branch

I am gonna close it for that reason as you will have to create a new PR in order to change the target branch

**Pull request closed without merging**